editors/vscode/esbuild.js

```diff
@@ -50,7 +50,7 @@ esbuild
         entryPoints: ["src/browser-lsp-worker.ts"],
         bundle: true,
         outfile: "out/browserServerMain.js",
-        format: "iife",
+        format: "esm",
```

Comment:
Are you sure about this one? `browserServerMain.js` becomes the source for a web worker, and we added the `iife` here because Firefox doesn't support web workers yet (or didn't at that point). But if you change to esm here, think you need to change the constructor call to `Worker`:

```
    const serverMain = Uri.joinPath(
        context.extensionUri,
        "out/browserServerMain.js",
    );

    const worker = new Worker(serverMain.toString(true));
```

and pass an options object that sets the type to module, like `{ type: "module" }`.

Reply:
Looks like the FireFox release in June this year fixes the missing ESM support. But I think the code on our side still needs to be adapted. https://caniuse.com/mdn-api_worker_worker_ecmascript_modules 

The JavaScript grammar does not allow for guessing whether the source is plain or ESM, so we must use the options object.

Reply:
Isn't `browserServerMain.js` used when the VS Code extension is used in VS Code for the web?

In any case, seems Firefox is fixed. But my point remains: I think you need to use the `Worker` constructor differently so that the browser knows what grammar to expect (es modules have a different entry point in the grammar).